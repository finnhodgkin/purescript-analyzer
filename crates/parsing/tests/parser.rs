macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
