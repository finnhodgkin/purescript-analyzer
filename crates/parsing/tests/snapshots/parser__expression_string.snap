---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..69
      ModuleHeader@0..31
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..24
          UPPER@7..23 "ExpressionString"
          Comment@23..24
            WHITESPACE@23..24 " "
        ModuleExportList@24..24
        WHERE@24..29 "where"
        Comment@29..31
          WHITESPACE@29..31 "\n\n"
      LAYOUT_START@31..31 ""
      ModuleImports@31..31
      ModuleStatements@31..69
        ValueEquation@31..48
          LOWER@31..37 "string"
          Comment@37..38
            WHITESPACE@37..38 " "
          EquationBinders@38..38
          Unconditional@38..48
            EQUAL@38..39 "="
            Comment@39..40
              WHITESPACE@39..40 " "
            WhereExpression@40..48
              ExpressionString@40..48
                STRING@40..47 "\"hello\""
                Comment@47..48
                  WHITESPACE@47..48 "\n"
        LAYOUT_SEPARATOR@48..48 ""
        ValueEquation@48..69
          LOWER@48..54 "string"
          Comment@54..55
            WHITESPACE@54..55 " "
          EquationBinders@55..55
          Unconditional@55..69
            EQUAL@55..56 "="
            Comment@56..57
              WHITESPACE@56..57 " "
            WhereExpression@57..69
              ExpressionString@57..69
                RAW_STRING@57..68 "\"\"\"hello\"\"\""
                Comment@68..69
                  WHITESPACE@68..69 "\n"
      LAYOUT_END@69..69 ""
      END_OF_FILE@69..69 ""
    ,
    [],
)
